test/micro/org/openjdk/bench/vm/compiler/VectorSubword.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.openjdk.bench.vm.compiler;
+
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.*;
+
+import java.util.concurrent.TimeUnit;
+import java.util.Random;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Thread)
+@Warmup(iterations = 4, time = 2, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 4, time = 2, timeUnit = TimeUnit.SECONDS)
+@Fork(value = 3)
+public class VectorSubword {
+    @Param({"1024"})
+    public int SIZE;
+
+    private byte[] bytes;
+    private short[] shorts;
+    private int[] ints;
+
+    @Setup
+    public void init() {
+        bytes = new byte[SIZE];
+        shorts = new short[SIZE];
+        ints = new int[SIZE];
+    }
+
+    @Benchmark
+    public void intToShort() {
+        for (int i = 0; i < SIZE; i++) {
+            shorts[i] = (short) ints[i];
+        }
+    }
+
+    @Benchmark
+    public void intToByte() {
+        for (int i = 0; i < SIZE; i++) {
+            bytes[i] = (byte) ints[i];
+        }
+    }
+
+    @Benchmark
+    public void shortToByte() {
+        for (int i = 0; i < SIZE; i++) {
+            bytes[i] = (byte) shorts[i];
+        }
+    }
+}
```

Comment:
Ah, these are all casting to smaller types. What about casting to larger types?

Reply:
Originally I thought that larger type conversions were only available on AVX-512 based on the ad-file, but reading it more carefully now I see that they are indeed supported for AVX! I'll go ahead and generalize the changes further to make casting to larger types supported as well. I think this leaves `char` as the only type not supported, which I can look at in a follow-up RFE.

Reply:
Well it's ok to leave the patch as is, and just have a follow-up RFE that improves things from here. It could make things easier to review.

Reply:
@jaskarth

Reply:
@eme64 I took a look last night and it looks like supporting casting to larger types (other than `char`) just involves changing the filtering logic in `matcher_x86.hpp` and adding more unit tests, without modifying any of the core logic. I think it could make sense to keep it in the same patch, what do you think? Either way would work with me.

Reply:
I leave it up to you. If it's not much then pack it in the same, else splitting is fine with me too ;)